core/commonMain/src/kotlinx/serialization/Serializers.kt

```diff
@@ -189,24 +189,38 @@ private fun SerializersModule.serializerByKTypeImpl(
     val isNullable = type.isMarkedNullable
     val typeArguments = type.arguments.map(KTypeProjection::typeOrThrow)
 
-    val cachedSerializer = if (typeArguments.isEmpty()) {
+    val cachedSerializer = if (this.hasInterfaceContextualSerializers) {
+        null
+    } else if (typeArguments.isEmpty()) {
         findCachedSerializer(rootClass, isNullable)
     } else {
-        findParametrizedCachedSerializer(rootClass, typeArguments, isNullable).getOrNull()
+        findParametrizedCachedSerializer(
+            rootClass,
+            typeArguments,
+            isNullable
+        ).getOrNull()
+    }
+    if (cachedSerializer != null) return cachedSerializer
+    var polymorphicFallback: KSerializer<Any?>? = null
+    if (rootClass.isInterface()) {
+        // Note that there is no check if interface is sealed, because for sealed interfaces we always have their serializers in cache
```

Comment:
Yes. Although this code still correct, because `rootClass.serializerOrNull()` will return proper serializer for sealed interfaces. I'll update the comment